fhir-core/src/main/java/com/ibm/fhir/core/util/ResourceTypeHelper.java

```diff
@@ -0,0 +1,122 @@
+        }
+    }
+
+    /**
+     * @return the set of resource type names that were either introduced in 4.3.0 (e.g. Ingredient) or changed
+     *          in backwards-incompatible ways in the 4.3.0 release (e.g. Evidence and EvidenceVariable)
+     */
+    public static Set<String> getNewOrBreakingResourceTypeNames() {
+        return R4B_ONLY_RESOURCES;
+    }
+
+    private static Set<ResourceTypeName> collectRemovedResourceTypes() {
+        Set<ResourceTypeName> set = new HashSet<>();
+        set.add(ResourceTypeName.EFFECT_EVIDENCE_SYNTHESIS);
+        set.add(ResourceTypeName.MEDICINAL_PRODUCT);
+        set.add(ResourceTypeName.MEDICINAL_PRODUCT_AUTHORIZATION);
+        set.add(ResourceTypeName.MEDICINAL_PRODUCT_CONTRAINDICATION);
+        set.add(ResourceTypeName.MEDICINAL_PRODUCT_INDICATION);
+        set.add(ResourceTypeName.MEDICINAL_PRODUCT_INGREDIENT);
+        set.add(ResourceTypeName.MEDICINAL_PRODUCT_INTERACTION);
+        set.add(ResourceTypeName.MEDICINAL_PRODUCT_MANUFACTURED);
+        set.add(ResourceTypeName.MEDICINAL_PRODUCT_PACKAGED);
+        set.add(ResourceTypeName.MEDICINAL_PRODUCT_PHARMACEUTICAL);
+        set.add(ResourceTypeName.MEDICINAL_PRODUCT_UNDESIRABLE_EFFECT);
+        set.add(ResourceTypeName.RISK_EVIDENCE_SYNTHESIS);
+        set.add(ResourceTypeName.SUBSTANCE_NUCLEIC_ACID);
+        set.add(ResourceTypeName.SUBSTANCE_POLYMER);
+        set.add(ResourceTypeName.SUBSTANCE_PROTEIN);
+        set.add(ResourceTypeName.SUBSTANCE_REFERENCE_INFORMATION);
+        set.add(ResourceTypeName.SUBSTANCE_SOURCE_MATERIAL);
+        set.add(ResourceTypeName.SUBSTANCE_SPECIFICATION);
+        return set;
+    }
+
+    private static Set<ResourceTypeName> collectR4bOnlyResourceTypes() {
+        Set<ResourceTypeName> set = new HashSet<>();
+        set.add(ResourceTypeName.ADMINISTRABLE_PRODUCT_DEFINITION);
+        set.add(ResourceTypeName.CITATION);
+        set.add(ResourceTypeName.CLINICAL_USE_DEFINITION);
+        set.add(ResourceTypeName.EVIDENCE_REPORT);
+        set.add(ResourceTypeName.INGREDIENT);
+        set.add(ResourceTypeName.MANUFACTURED_ITEM_DEFINITION);
+        set.add(ResourceTypeName.MEDICINAL_PRODUCT_DEFINITION);
+        set.add(ResourceTypeName.NUTRITION_PRODUCT);
+        set.add(ResourceTypeName.PACKAGED_PRODUCT_DEFINITION);
+        set.add(ResourceTypeName.REGULATED_AUTHORIZATION);
+        set.add(ResourceTypeName.SUBSCRIPTION_STATUS);
+        set.add(ResourceTypeName.SUBSCRIPTION_TOPIC);
+        set.add(ResourceTypeName.SUBSTANCE_DEFINITION);
+        // The following resource types existed in R4, but have breaking changes in R4B.
+        // Because we only support the R4B version, we don't want to advertise these in our 4.0.1 statement.
+        set.add(ResourceTypeName.DEVICE_DEFINITION);
+        set.add(ResourceTypeName.EVIDENCE);
+        set.add(ResourceTypeName.EVIDENCE_VARIABLE);
+        // TODO: make final decision on whether to lump these in with the breaking resources
```

Comment:
resolution of todo?

Reply:
I think its gonna need to wait until HL7 indicates whether they will back out the breaking change on these resource types or not.  Discussion at https://chat.fhir.org/#narrow/stream/179166-implementers/topic/R4B.20compatibilty/near/269414612 seemed to fizzle.

If the back out the changes, we just need to remove these 3 lines.
If they don't back them out, we may still want to server these resource types when the request is for fhirVersion 4.0, even though there's a chance that some unsuspecting R4 client could break. That would require team discussion I think.

Reply:
Based on the feedback on the sprint playback, I'm thinking we should advertise support for PlanDefinition and Activity in our `4.0` mode, even if HL7 *doesn't* back out the new choice types.
I can cover the concerns with that with the doc for https://github.com/IBM/FHIR/issues/3280